pyscriptjs/src/components/pyrepl.ts

```diff
@@ -58,8 +58,8 @@ export function make_PyRepl(interpreter: InterpreterClient, app: PyScriptApp) {
             this.setAttribute('src', value);
         }
         attributeChangedCallback(name: string, oldVal: string, newVal: string) {
-            if(name==="src"&&newVal!==oldVal){
-                this.src=newVal;
+            if (name === 'src' && newVal !== oldVal) {
+                this.src = newVal;
```

Comment:
you are triggering the setter here which in turns triggers the `attributeChangedCallback` ... you don't need to do so because when this callback happens the `src` is already `newVal` ... moreover, if you don't add the static `observedAttributes` in the class this won't work if something else changes the attribute in the wild.

Reply:
Modify it like this?
```js
attributeChangedCallback(name: string, oldVal: string, newVal: string) {
    void this.loadReplSrc();
}
```

Reply:
the `void` is redundant there too, but basically yes, as long as you add `static observedAttributes = ['src']` to the class ... if you don't do that, this callback is never triggered https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_custom_elements#using_the_lifecycle_callbacks

Reply:
However, if I do not add a void, an error will be reported:
> ESLint: Promises must be awaited, end with a call to .catch, end with a call to .then with a rejection handler or be explicitly marked as ignored with the `void` operator.(@typescript-eslint/no-floating-promises)

Reply:
I see ... then I believe if we want to agree with *eslint* we need to make `attributeChangedCallback` an `async` callback and `await` that internal call ... here we're just bypassing an error without awaiting anything explicitly ... basically the linter has a reason to complain and we need to escape that reason with misleading details ... let's keep it as is for the time being, but it's awkward to reason about that `void` (to me).